__tests__/integration/helpers/wallet.helper.js

```diff
@@ -437,3 +438,49 @@ export async function waitNextBlock(storage) {
     throw new Error('Timeout reached when waiting for the next block.');
   }
 }
+
+/**
+ * This method awaits a tx to be confirmed by a block and then resolves the promise.
+ *
+ * It does not return any content, only delivers the code processing back to the caller at the
+ * desired time.
+ *
+ * @param {HathorWallet} hWallet
+ * @param {String} txId
+ * @param {number | null | undefined} timeout
+ * @returns {Promise<void>}
+ */
+export async function waitTxConfirmed(hWallet, txId, timeout) {
+  // Only return the positive response after the tx has a first block
+  return new Promise(async (resolve, reject) => {
+    let timeoutHandler;
+    if (timeout) {
+      // Timeout handler
+      timeoutHandler = setTimeout(async () => {
+        reject(new Error(`Timeout of ${timeout}ms without confirming the transaction`));
+      }, timeout);
+    }
+
+    while (await getTxFirstBlock(hWallet, txId) === null) {
+      await delay(1000);
+    }
```

Comment:
this might throw, maybe wrap with a try/catch and use the reject here as well

Reply:
Done https://github.com/HathorNetwork/hathor-wallet-lib/pull/604/commits/a0d1a92a6207778791a75fcb40ccd0029e4c92a2